tests/specs/packages.js

```diff
@@ -0,0 +1,126 @@
+define(["dist/amd-loader"], function(AMDLoader) {
+  describe("Test suite", function() {
+    var bitloader;
+    beforeEach(function() {
+      bitloader = new AMDLoader({
+        baseUrl: "../",
+        packages: [
+          {
+            location: "tests",
+            main: "main",
+            name: "js"
+          }
+        ]
+      });
+    });
+
+    describe("When Importing a number", function() {
+      var number;
+      beforeEach(function() {
+        return bitloader.import("js/number.js")
+          .then(function (_number) {
+            console.log(arguments);
+            number = _number;
+          });
+      });
+
+      it("then the number is 3.14", function() {
+        expect(number).to.equal(3.14);
+      });
+    });
+
+    describe("When Importing an empty object", function() {
+      var result;
+      beforeEach(function() {
+        return bitloader.import("js/emptyobject.js")
+          .then(function (_result) {
+            result = _result;
+          });
+      });
+
+      it("then the result is an object", function() {
+        expect(result).to.be.an("object");
+      });
+
+      it("then the result is an object with no properties", function() {
+        expect(Object.keys(result).length).to.equal(0);
+      });
+    });
+
+    describe("When Importing two files", function() {
+      var emptyobject, emptystring;
+      beforeEach(function () {
+        return bitloader.import(["js/emptyobject", "js/emptystring"])
+          .then(function (_emptyobject, _emptystring) {
+            emptyobject = _emptyobject;
+            emptystring = _emptystring;
+          });
+      });
+
+      it("then the first result is one object", function() {
+        expect(emptyobject).to.be.an("object");
+      });
+
+      it("then the first result is one object with no properties", function() {
+        expect(Object.keys(emptyobject).length).to.equal(0);
+      });
+
+      it("then the second result is a string", function() {
+        expect(emptystring).to.be.a("string");
+      });
+
+      it("then the second result is an empty string", function () {
+        expect(emptystring.length).to.equal(0);
+      });
+    });
+
+    describe("When Importing nested dependencies in deep3", function() {
+      var result;
+      beforeEach(function () {
+        return bitloader.import("js/deep3")
+          .then(function (_result) {
+            result = _result;
+          });
+      });
+
+      it("then deep3 is an object", function() {
```

Comment:
Here we can wrap this with a `describe` with text that describes what's happening.  One thing that's even harder than naming variables is coming up with text for unit tests.  I would call this something like `describe("and deep3 is loaded", function() {});`

Then each `expect` below is wrapped in a single `it` describing what the test is doing.  For example, you can have something like:

``` javascript
it("deep3 is an object", function() {
  expect(result).to.be.an("object");
});

it("deep3 has only one property", function() {
  expect(Object.keys(result).length).to.equal(1);
});

it("deep2 is an object", function() {
  expect(result.deep2).to.be.an("object");
});
```

The idea is to have readable output that gives you an idea of what is being tested.

One thing I just noticed is that the data in `deep3.js` is a bit wrong.

``` javascript
Bitloader.define(["js/deep2"], function(deep2) {
  return {
    "deep3": deep2
  };
});
```

should really be 

``` javascript
Bitloader.define(["js/deep2"], function(deep2) {
  return {
    "deep2": deep2
  };
});
```

Those `deep` module can probably use a little adjustment to make more readable tests.